src/main/java/com/exadel/movie_new/service/MovieServiceImpl.java

```diff
@@ -0,0 +1,86 @@
+package com.exadel.movie_new.service;
+
+import com.exadel.movie_new.model.Movie;
+import com.google.gson.Gson;
+import com.google.gson.reflect.TypeToken;
+import org.springframework.stereotype.Service;
+
+import java.io.FileWriter;
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+
+@Service
+public class MovieServiceImpl implements MovieService {
+
+
+    @Override
+    public Movie addMovie(Movie movie) {
+        List<Movie> movieData = readJsonData();
+        Movie moviePresent = null;
+
+        if (movieData != null) {
+            moviePresent = movieData.parallelStream().filter(mov -> mov.getId().equals(movie.getId())).findAny().orElse(null);

```

Comment:
Just FYI. In real life we never set ID manually. It always doing by DB. So this filtering will be redundant. But for now, as we use file as storage - it's ok.